src/sentry/tasks/commit_context.py

```diff
@@ -44,19 +45,35 @@ def queue_comment_task_if_needed(
         extra={"organization_id": commit.organization_id, "merge_commit_sha": commit.key},
     )
 
+    # client will raise ApiError if the request is not successful
     response = installation.get_client().get_pullrequest_from_commit(repo=repo.name, sha=commit.key)
 
-    if not (response.status_code == 200 and isinstance(response, list) and len(response) == 1):
+    if not (isinstance(response, list) and len(response) == 1):
         # the response should return a single PR, return if multiple
+        if len(response) > 1:
```

Comment:
This check was in the previous PR too right?
I think i mentioned as a non blocking nit comment in the other PR - but maybe lets move the multiple PRs check validation into the `get_pullrequest_from_commit` method instead and either return a single PR - or None/raise

Reply:
i'm iffy about moving the check into the actual client because we don't do any checks when we hit other github APIs. there is always SOME response returned from the API, but here we're only executing further logic if it returns a single PR.

i feel like this is a special case so i can clarify the multiple possible responses in the docstring in the mixin function